projects/ngx-datatable/src/lib/components/body/body.component.ts

```diff
@@ -64,145 +67,181 @@ import {
-              [verticalScrollVisible]="verticalScrollVisible"
-              (treeAction)="onTreeAction(group)"
-              (activate)="selector.onActivate($event, indexes.first + i)"
-              (drop)="drop($event, group, rowElement)"
-              (dragover)="dragOver($event, group)"
-              (dragenter)="dragEnter($event, group, rowElement)"
-              (dragleave)="dragLeave($event, group, rowElement)"
-              (dragstart)="drag($event, group, rowElement)"
-              (dragend)="dragEnd($event, group)"
+              [rowIndex]="getRowIndex(group && group[i])"
+              [selected]="selected"
+              (rowContextmenu)="rowContextmenu.emit($event)"
             >
-            </datatable-body-row>
-          </ng-template>
-
-          <ng-container *ngIf="isGroup(group)">
-            <!-- The row typecast is due to angular compiler acting weird. It is obvious that it is of type TRow, but the compiler does not understand. -->
-            <datatable-body-row
+              @if (rowDefTemplate) {
+                <ng-container
+                  *rowDefInternal="{
+                    template: rowDefTemplate,
+                    rowTemplate: bodyRow,
+                    row: group,
+                    index: i
+                  }"
+                />
+              } @else {
+                @if (isRow(group)) {
+                  <datatable-body-row
+                    role="row"
+                    tabindex="-1"
+                    #rowElement
+                    [disable$]="rowWrapper.disable$"
+                    [isSelected]="selector.getRowSelected(group)"
+                    [innerWidth]="innerWidth"
+                    [offsetX]="offsetX"
+                    [columns]="columns"
+                    [rowHeight]="getRowHeight(group)"
+                    [row]="group"
+                    [rowIndex]="getRowIndex(group)"
+                    [expanded]="getRowExpanded(group)"
+                    [rowClass]="rowClass"
+                    [displayCheck]="displayCheck"
+                    [treeStatus]="group?.treeStatus"
+                    [ghostLoadingIndicator]="ghostLoadingIndicator"
+                    [draggable]="rowDraggable"
+                    [verticalScrollVisible]="verticalScrollVisible"
+                    (treeAction)="onTreeAction(group)"
+                    (activate)="selector.onActivate($event, indexes.first + i)"
+                    (drop)="drop($event, group, rowElement)"
+                    (dragover)="dragOver($event, group)"
+                    (dragenter)="dragEnter($event, group, rowElement)"
+                    (dragleave)="dragLeave($event, group, rowElement)"
+                    (dragstart)="drag($event, group, rowElement)"
+                    (dragend)="dragEnd($event, group)"
+                  >
+                  </datatable-body-row>
+                }
+              }
+
+              <ng-template #bodyRow>
```

Comment:
If we need this template, should we than reuse it in line 117. Feels strange to have the same element here twice.